### ISSUE TITLE
autocomplete broken in tsserver 2.6

### ISSUE THREAD

**Issue by prabirshrestha** [theia-ide/typescript-language-server#20]
Here is the log from vim-lsp.

```
11/5/2017 10:19:33 AM:["<---",1,"typescript-language-server",{"response":{"id":9,"jsonrpc":"2.0","error":{"code":-32603,"message":"Request textDocument/completion failed with message: Error processing request. Cannot read property 'kind' of undefined\nTypeError: Cannot read property 'kind' of undefined\n    at Object.isClassElement (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:9931:25)\n    at isFromClassElementDeclaration (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:65915:27)\n    at tryGetClassLikeCompletionContainer (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:65943:33)\n    at tryGetGlobalSymbols (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:65560:42)\n    at getCompletionData (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:65478:22)\n    at Object.getCompletionsAtPosition (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:65060:34)\n    at Object.getCompletionsAtPosition (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:79777:35)\n    at IOSession.Session.getCompletions (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:86199:64)\n    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:85311:61)\n    at d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:86669:88\n    at IOSession.Session.executeWithRequestId (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:86660:28)\n    at IOSession.Session.executeCommand (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:86669:33)\n    at IOSession.Session.onMessage (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:86689:35)\n    at Interface.<anonymous> (d:\\tmp\\node_modules\\typescript\\lib\\tsserver.js:87881:27)\n    at emitOne (events.js:115:13)\n    at Interface.emit (events.js:210:7)\n    at Interface._onLine (readline.js:282:10)\n    at Interface._normalWrite (readline.js:424:12)\n    at Socket.ondata (readline.js:141:10)\n    at emitOne (events.js:115:13)\n    at Socket.emit (events.js:210:7)\n    at addChunk (_stream_readable.js:264:12)\n    at readableAddChunk (_stream_readable.js:251:11)\n    at Socket.Readable.push (_stream_readable.js:209:10)\n    at Pipe.onread (net.js:587:20)"}},"request":{"method":"textDocument/completion","jsonrpc":"2.0","id":9,"params":{"textDocument":{"uri":"file:///D:/tmp/file.ts"},"position":{"character":17,"line":122}}}}]
```

Here is the formatted stack trace.

```
Request textDocument/completion failed with message: Error processing request. Cannot read property 'kind' of undefined
TypeError: Cannot read property 'kind' of undefined
    at Object.isClassElement (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:9931:25)
    at isFromClassElementDeclaration (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:65915:27)
    at tryGetClassLikeCompletionContainer (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:65943:33)
    at tryGetGlobalSymbols (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:65560:42)
    at getCompletionData (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:65478:22)
    at Object.getCompletionsAtPosition (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:65060:34)
    at Object.getCompletionsAtPosition (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:79777:35)
    at IOSession.Session.getCompletions (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:86199:64)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:85311:61)
    at d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:86669:88
    at IOSession.Session.executeWithRequestId (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:86660:28)
    at IOSession.Session.executeCommand (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:86669:33)
    at IOSession.Session.onMessage (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:86689:35)
    at Interface.<anonymous> (d:\\tmp\node_modules\\typescript\\lib\\tsserver.js:87881:27)
    at emitOne (events.js:115:13)
    at Interface.emit (events.js:210:7)
    at Interface._onLine (readline.js:282:10)
    at Interface._normalWrite (readline.js:424:12)
    at Socket.ondata (readline.js:141:10)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at addChunk (_stream_readable.js:264:12)
    at readableAddChunk (_stream_readable.js:251:11)
    at Socket.Readable.push (_stream_readable.js:209:10)
    at Pipe.onread (net.js:587:20)"}}
```
I have been unblocking myself by using ts 2.5.3

Seems like even others are having issues: https://github.com/mhartington/nvim-typescript/issues/89

**Comment by svenefftinge** [comment]
The tests fail, too, with 2.6.1. The tsserver log includes the following exceptions:
```
Err 22    [9:39:56.940] Exception on executing command {"command":"open","seq":5,"type":"request","arguments":{"file":"/Users/efftinge/Documents/theia-dev/typescript-language-server/test-data/bar.ts","fileContent":"\n        export function foo(): void {\n          unknown('test')\n        }\n      "}}:

    Debug Failure. False expression.

    Error: Debug Failure. False expression.
    at ProjectService.getOrCreateScriptInfoForNormalizedPath (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:84573:30)
    at ProjectService.getOrCreateScriptInfoOpenedByClientForNormalizedPath (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:84548:29)
    at ProjectService.openClientFileWithNormalizedPath (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:84692:33)
    at IOSession.Session.openClientFile (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:86030:37)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:85262:31)
    at /Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:86669:88
    at IOSession.Session.executeWithRequestId (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:86660:28)
    at IOSession.Session.executeCommand (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:86669:33)
    at IOSession.Session.onMessage (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:86689:35)
    at Interface.<anonymous> (/Users/efftinge/Documents/theia-dev/typescript-language-server/node_modules/typescript/lib/tsserver.js:87881:27)
    at emitOne (events.js:115:13)
```

**Comment by svenefftinge** [comment]
I fixed an issue regarding multiple 'open' calls for the same document and published 0.1.5.
Could you verify if that also solves your issue?

**Comment by prabirshrestha** [comment]
Seems to work. Thanks.
